src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php

```diff
@@ -63,8 +63,18 @@ protected function createController($controller)
                 list($service, $method) = explode(':', $controller, 2);
 
                 return array($this->container->get($service), $method);
-            } elseif ($this->container->has($controller) && method_exists($service = $this->container->get($controller), '__invoke')) {
-                return $service;
+            } elseif ($this->container->has($controller)) {
+                $controllerInstance = $this->container->get($controller);
+                if (!method_exists($controllerInstance, '__invoke')) {
+                    $exceptionMessage = sprintf(
+                        'The class "%s" specified as controller service "%s" must define an __invoke method, but does not.',
+                        get_class($controllerInstance),
+                        $controller
+                    );
+                    throw new \LogicException($exceptionMessage);
```

Comment:
I suggest removing the $exceptionMessage tempvar and put the message on one single (long) line. That's what we do elsewhere is the code base.